src/accumulator/accumulated_map.rs

```diff
@@ -50,6 +50,13 @@ impl AccumulatedMap {
             .unwrap()
             .extend_with_accumulated(output);
     }
+
+    pub(crate) fn take(&mut self) -> Self {
+        Self {
+            map: self.map.drain().collect(),
+            inputs: self.inputs,
+        }
+    }
```

Comment:
This looks wrong. We should use `std::mem::take` on `self.inputs`

Reply:
Oops, good catch, though this was a change I've done after seeing the rebase change